x-pack/plugins/maps/public/classes/sources/es_geo_grid_source/render_as_select.tsx

```diff
@@ -30,21 +32,17 @@ export function RenderAsSelect(props: {
   onChange: (newValue: RENDER_AS) => void;
   isColumnCompressed?: boolean;
 }) {
+  const currentOption = options.find((option) => option.value === props.renderAs) || options[0];
+  const [selectedOption, setSelectedOption] = useState(currentOption.id);
```

Comment:
nit: none of the React-code in the `maps/*` plugin, so I'd be hesitant to introduce now.